website/docs: Apple Business Manager Integration.

GirlBossRush opened the pull request:
This PR adds an integration guide for Apple Business Manager.

 - Fixes missing required attribute.
 - Adjusts visible headers in table of contents.
 - Fixes Docusaurus warning for nested paragraphs.


---

## Checklist

-   [ ] Local tests pass (`ak test authentik/`)
-   [ ] The code has been formatted (`make lint-fix`)

If an API change has been made

-   [ ] The API schema has been updated (`make gen-build`)

If changes to the frontend have been made

-   [ ] The code has been formatted (`make web`)

If applicable

-   [x] The documentation has been updated
-   [x] The documentation has been formatted (`make website`)

dominic-r commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,285 @@
+authentik_preview: true
+---
+
+## What is Apple Business Manager?
```
Shouldn't the [format mentioned in the template](https://github.com/goauthentik/authentik/blob/7e7fc75e7739c988dd64f92785baae54a975269e/website/integrations/template/service.md?plain=1#L7) be respected? So, putting this section in quote block and adding a link to the ABM website?

dominic-r commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,285 @@
+
+The following placeholders are used:
+
+- `authentik.company`: The Fully Qualified Domain Name of the authentik installation.
```
In the template, we use FQDN. Is this something we want to change in all existing documentation as well?

tanberry replied:
For now let's leave it as FDQN,please, to match the other dos. Then we can decide if we want to do a separate massive PR to go across the whole docs set, and maybe say "FDQN (Fully Qualified Domain Name)...". I think most users will know what FDQN is, though. But yes, for now please let's use the acronym, like in the other docs.

dominic-r commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,285 @@
+
+### Scope Mapping
+
+From the Admin interface, navigate to **Customization** -> **Property Mappings**.
```
IIRC, tana prefers "In" rather than "From"

dominic-r commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,285 @@
+
+:::tip{title="OpenID Configuration URL"}
+
+Before leaving the provider overview page, make sure to copy the _OpenID Configuration URL_ value to your text editor. We'll need this when configuring Apple Business Manager.
```
Shouldn't a sentence or two be added to tell the user how to reach the provider overview page?

dominic-r commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,285 @@
+5. `authentik default OAuth Mapping: OpenID 'openid'`
+6. `authentik default OAuth Mapping: OpenID 'offline_access'`
+
+Visually confirm that you've copied the _Client ID_ and _Client Secret_ before pressing **Finish**. You'll need these values when configuring Apple Business Manager. Save the provider and confirm that the new provider is listed in the provider overview.
```
Since Client ID and Client Secret are values in the UI, they should be put in bold as per the [styling section lf the dev docs](https://docs.goauthentik.io/docs/developer-docs/docs/style-guide#fonts-and-font-styling)

dominic-r commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,285 @@
+
+Choosing to capture all accounts is an irreversible action which affects all users with an email address in the federated domain, regardless of their enrollment status.
+
+**Once captured, the accounts can't be reverted to personal Apple IDs.**
```
You mention on the previous line that the action is irreversible. You could maybe merge this and the previous line just to keep things concise?

dominic-r commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,285 @@
+| Client ID         | `[Your Client ID]`      |
+| Client Secret     | `[Your Client Secret]`  |
+| SSF Config URL    | `[Your SSF Config URL]` |
+| OpenID Config URL | `[Your OpenID Config]`  |
```
```suggestion
| OpenID Config URL | `[Your OpenID Config URL]`  |
```

dominic-r reviewed (commented):
Hi Teffen. Here are a few things I noticed (and the Codespell warnings). Looks like a very complete documentation overall!

tanberry commented on website/docs/add-secure-apps/providers/ssf/index.md:
```diff
@@ -1,30 +1,39 @@
+Shared Signals Framework (<abbr>SSF</abbr>) is a collection of standards for sharing asynchronous security signals and events across multiple applications and an identity provider. SSF leverages the APIs of the application and the IdP, using privacy-protected, secure webhooks.
 
-## About Shared Signals Framework
+## What are security signals?
```
While that title is good, I want to leaveit as is, with the word "About" as a cognitive hint that this is the conceptual section... it's in the template for same reason. (Though yes, the proposed title also is clearly conceptual. I'd just like to stay consistent with our headers/titles.)

tanberry commented on website/docs/add-secure-apps/providers/ssf/index.md:
```diff
@@ -1,30 +1,39 @@
+An SSF provider allows applications to subscribe to _security signals_, translating them into Security Event Tokens (<abbr>SET</abbr>) that are captured by the IdP. In this arrangement, authentik acts as an mediator from an external event _transmitter_, ultimately with a specific application acting as the _receiver_ of the events.
 
-Events in authentik that are tracked via SSF include when an MFA device is added or removed, logouts, sessions being revoked by Admin or user clicking logout, or credentials changed.
+### Transmitter and receiver
```
Thanks for the new section, well-written! My only concern is that, if I understand correctly, there are cases where this can be reversed, and the App is the transmitter. That's why I have phrases like on line 13... "in authentik"... to be clear that we are describing how authentik works, not generalizing.

tanberry commented on website/docs/add-secure-apps/providers/ssf/index.md:
```diff
@@ -1,30 +1,39 @@
-Another example use case is when an application uses SSF to subscribe to authorization events because the application needs to know if a user changed their password in authentik. If a user did change their password, then the application receives a POST request to write the fact that the password was changed.
-
-## About using SSF in authentik
+Another example use case is when an application uses SSF to subscribe to authorization events because the application needs to know if a user changed their password in authentik. If a user changes their password, then the application will receive a POST request to relay an event that the password was changed.
```
Is event is for sure the correct term here...? I want to be careful with our use of the word event (versus signal, and also versus our other types of events).

tanberry commented on website/docs/add-secure-apps/providers/ssf/index.md:
```diff
@@ -1,30 +1,39 @@
+## SSF in authentik
 
-The SSF provider in authentik serves as a [backchannel provider](../../applications/manage_apps#backchannel-providers). Backchannel providers are used to augment the functionality of the main provider for an application. Thus you will still need to [create a typical application/provider pair](../../applications/manage_apps#instructions) (using an OIDC provider), and when creating the application, ...
+SSF providers are configured in authentik by creating a [backchannel provider](../../applications/manage_apps#backchannel-providers). Backchannel providers are used to augment the functionality of the main provider for an application. Thus, you will still need to [create a typical application/provider pair](../../applications/manage_apps#instructions) (using an OIDC provider), and when creating t ...
```
I find his sentence confusing/too vague. The SSF provider *is* the backchannel provider. Assuming that is accurate, I want to be as clear as possible about that. Your rewording implies that one creates the SSF provider, and then also configure an additional entity, the backchannel provider.

tanberry commented on website/docs/add-secure-apps/providers/ssf/index.md:
```diff
@@ -36,12 +45,19 @@ Note that authentik doesn't specify which events to subscribe to; instead the ap
-Optionally, you can specify a event retention time period: this value determines how long events are stored for. If an event could not be sent correctly, and retries occur, the event's expiration is also increased by this duration.
+You may also specify a event retention time period. If an event could not be sent correctly, and retries occur, the event's expiration is also increased by this duration.
+
+:::info{title="Events and SETs"}
```
I'm glad to know about the ability to give a specific title to an info block, but I don't want to use one here. Especially not this title, becaue it makes it sound like the info block is only about SSF Ents and SETs, when in reality it is pointing out that ther are other types of events, unrelated to these SSF events.

tanberry replied:
Also we both missed "an" versus "a" before "event". :-)

tanberry commented on website/docs/add-secure-apps/providers/ssf/index.md:
```diff
@@ -36,12 +45,19 @@ Note that authentik doesn't specify which events to subscribe to; instead the ap
-Be aware that the SET events are different events than those displayed in the authentik Admin interface under **Events**.
 :::
+
+## External references
```
We have a link (or I did) to the spec up above, in the first paragraph. I'd like to leave it there, where we describe what SSF is. Best practice is to link in-line, exactly where the reader will wants the info, and exactly where we reference it.

I'm not ready to commit to adding this new section to new docs, nor all of the docs (we need to be consistent)... nor to the work of maintaining them. If we were 10-person Docs team, maybe, but we need to keep it lean and mean and easy to maintain going forward.

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+
+- `authentik.company`: The Fully Qualified Domain Name of the authentik installation.
+
+## authentik Configuration
```
```suggestion
## authentik configuration
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+
+## authentik Configuration
+
+Before you can proceed with the Apple Business Manager configuration, you'll need to configure authentik to act as an identity provider for Apple. Start by logging into your authentik instance and navigating to the Admin interface.
```
```suggestion
Before you proceed with the Apple Business Manager configuration, you will need to configure authentik to act as an identity provider for Apple. Start by logging into your authentik instance and navigating to the Admin interface.
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+
+Before you can proceed with the Apple Business Manager configuration, you'll need to configure authentik to act as an identity provider for Apple. Start by logging into your authentik instance and navigating to the Admin interface.
+
+### Scope Mapping
```
```suggestion
### Scope mapping
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+
+### Scope Mapping
+
+In the Admin interface, navigate to **Customization** -> **Property Mappings**.
```
Let's number the steps in each section...

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+
+In the Admin interface, navigate to **Customization** -> **Property Mappings**.
+
+Press **Create** and when prompted for a choice of mapping, select **Scope Mapping**.
```
We use "Click" in all other docs. Sorry, but let's be consistent for now and we can decide whether we want to change across the docs set later.

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+
+Press **Create** and when prompted for a choice of mapping, select **Scope Mapping**.
+
+#### Apple Business Manager Profile
```
Is this a user's profile? Or an account (like an org, or sevice account) profile/ If a user account, let's say "Apple Business Manager user profile".  (Remember our Style Guide calls for sentence-style capitalization for titles.)

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+
+Create a new scope mapping to handle the user's profile information:
+
+| Field      | Value                          |
```
Again for consistency reasons, I'd rather not use a table for only two values. Yet another topic we can discuss separately, for longer lists of values.

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+| Name       | Apple Business Manager profile |
+| Scope name | `profile`                      |
+
+Use the following expression to map the user's profile information:
```
Where do they enter this expression? Is it a field labeled "Expression"? If so, let's state that... we want our procedural docs, our steps, to be written about the UI, referencing the UI itself as much as possible, to help them follow the steps more easily. So if this is a field, then say something like "3. To map the user's profile informatin between x and y, add the following code to the **Expression** field."

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+
+Save the scope mapping and press "Create" to continue to the next scope mapping.
+
+#### Read Scope
```
```suggestion
#### Read the scope mapping
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+
+#### Read Scope
+
+Next, create a new scope mapping to handle read access to the user's profile information:
```
```suggestion
1. Next, create a new scope mapping to handle read access to the user's profile information:
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+
+:::tip{title="Keep your text editor ready"}
+
+authentik will automatically generate the _Client ID_ and _Client Secret_ values for the new provider.
```
```suggestion
authentik will automatically generate the `Client ID`  and `Client Secret` values for the new provider.
```

tanberry replied:
We want values to be in code font, not italics. Italics are reserved for variables. ;-)

tanberry replied:
Also need to have numbered steps, please

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+5. `authentik default OAuth Mapping: OpenID 'openid'`
+6. `authentik default OAuth Mapping: OpenID 'offline_access'`
+
+Visually confirm that you've copied the **Client ID** and **Client Secret** before pressing **Finish**. You'll need these values when configuring Apple Business Manager. Save the provider and confirm that the new provider is listed in the provider overview.
```
```suggestion
Visually confirm that you've copied the `Client ID` and `Client Secret` values before clicking **Finish**. You'll need these values when configuring Apple Business Manager. Save the provider and confirm that the new provider is listed in the provider overview.
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+
+:::tip{title="OpenID Configuration URL"}
+
+Before leaving the provider overview page, make sure to copy the **OpenID Configuration URL** value to your text editor. You'll need this when configuring Apple Business Manager.
```
```suggestion
Before leaving the provider overview page, make sure to copy the `OpenID Configuration URL` value to your text editor. You'll need this when configuring Apple Business Manager.
```

tanberry replied:
This one could go either way... if you were referring to the field label itself, the UI component, then bold it. BUt since you are talking about the alue related to that field, it's a value, so code font it is.

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+
+Before leaving the provider overview page, make sure to copy the **OpenID Configuration URL** value to your text editor. You'll need this when configuring Apple Business Manager.
+
+You can always find your provider's generated URLs by navigating to **Providers**, then selecting the provider by name. The URLs are listed under the **Overview** tab.
```
```suggestion
You can always find your provider's generated URLs by navigating to **Providers** and then selecting the provider by name. The URLs are listed under the **Overview** tab.
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+
+### Shared Signals Framework Provider
+
+While the OAuth2/OpenID Provider handles the authentication flow, you'll need to create a Shared Signals Framework Provider to handle the backchannel communication between authentik and Apple Business Manager.
```
```suggestion
While the OAuth2/OpenID provider handles the authentication flow, you'll need to create a Shared Signals Framework provider to handle the backchannel communication between authentik and Apple Business Manager.
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+| ----- | -------------------------- |
+| Name  | apple-business-manager-ssf |
+
+Press **Finish** to create the Shared Signals Framework Provider.
```
```suggestion
Click **Finish** to create the Shared Signals Framework Provider.
```

tanberry replied:
There are more values than just the **Name** to define, when creating an SSF provider, right? I think it would be best to simply link to the docs on how to create an SSF provider, instead of duplicating those steps here. Thoughts?

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+
+## What is Apple Business Manager?
+
+> Apple Business Manager is a web-based portal for IT administrators, managers, and procurement professionals to manage devices, and automate device enrollment.
```
Can we explain the relationship between ABM, ASM, and ABE? Is this Integration guide for all three? Or does one only integrate ABM, and then the others don't need any configuration, but work off the ABM config?

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+
+Once again, save the scope mapping and press "Create" to continue to create the final scope mapping.
+
+#### Manage Scope
```
Can we make this title more descriptive? Also, titles use sentence-style capitalization. ;-)

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+
+Finally, save the scope mapping and visually confirm the creation of all three scope mappings.
+
+### OAuth2/OpenID Provider
```
```suggestion
### OAuth2/OpenID provider
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+
+:::
+
+### Shared Signals Framework Provider
```
```suggestion
### Shared Signals Framework provider
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+
+While the OAuth2/OpenID Provider handles the authentication flow, you'll need to create a Shared Signals Framework Provider to handle the backchannel communication between authentik and Apple Business Manager.
+
+Once more from the authentik Admin interface, navigate to **Applications** -> **Providers** and press **Create**. This time, select **Shared Signals Framework Provider** and continue with the following form values:
```
```suggestion
From the authentik Admin interface, navigate to **Applications** -> **Providers** and click **Create**. This time, select **Shared Signals Framework Provider** and continue with the following form values:
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+
+Press **Finish** to create the Shared Signals Framework Provider.
+
+### Application Configuration
```
```suggestion
### Application configuration
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+
+### Application Configuration
+
+You're now ready to create the application configuration for Apple Business Manager. From the authentik Admin interface, navigate to **Applications** -> **Applications** and press **Create**.
```
Typically in our integration guides, we have two mai sections: authentik configuration, and application configuration.

To keep that (simple understandable) structure, let's move this content under the H2, under **## Apple Business Manager Configuration** section.

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+| Provider             | Apple Business Manager       |
+| Backchannel Provider | `apple-business-manager-ssf` |
+
+With the application created, you're almost ready to switch over to Apple Business Manager.
```
```suggestion
With the application created, you're almost ready to switch over to configuring Apple Business Manager.
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+
+:::tip{title="SSF Config URL"}
+
+Assigning an SSF provider to a application will generate _SSF Config URL_ value. You'll need this when configuring Apple Business Manager, so make sure navigate to **Providers** -> **Apple Business Manager SSF** and copy the _SSF Config URL_ value to your text editor.
```
```suggestion
Assigning an SSF provider to a application will generate `SSF Config URL` value. You'll need this when configuring Apple Business Manager, so make sure navigate to **Providers** -> **Apple Business Manager SSF** and copy the `SSF Config URL` value to your text editor.
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+
+---
+
+## Apple Business Manager Configuration
```
```suggestion
## Apple Business Manager configuration
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+    - And noted the _Client ID_, _Client Secret_, and _OpenID Configuration URL_.
+3. Created a Shared Signals Framework Provider to handle the backchannel communication between authentik and Apple Business Manager.
+    - Noted the _SSF Config URL_.
+4. Created application configuration for Apple Business Manager.
```
This is of course accurate, but how can we word it to say that we "created a new application in authentik for ABM". This is the way our docs and UI present this... it's not so much "configuring an application" but rather creating, and then configuring an application.

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+    - Noted the _SSF Config URL_.
+4. Created application configuration for Apple Business Manager.
+
+With these prerequisites in place, authentik is primed to act as an identity provider for Managed Apple Accounts.
```
```suggestion
With these prerequisites in place, authentik is ready to act as an identity provider for Managed Apple Accounts.
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+
+:::info{title="Managed Apple Accounts"}
+
+Similar to a user's personal Apple account, a _Managed Apple Account_ is created and managed by an organization, and can be used to access Apple services and devices. Both use an email address as the user name, but a _Managed Apple Account_ is owned by the organization, not the user.
```
```suggestion
Similar to a user's personal Apple account, a _Managed Apple Account_ is created and managed by an organization, and can be used to access Apple services and devices. Both use an email address as the user name, but a _Managed Apple Account_ is owned by the organization, not the user.
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+
+#### Lock your domain
+
+Upon successful domain verification, you'll be asked to confirm that you wish to lock your domain. Locking your domain ensures that only your organization can use your domain for federated authentication.
```
```suggestion
Upon successful domain verification, you are asked to confirm that you wish to lock your domain. Locking your domain ensures that only your organization can use your domain for federated authentication.
```

tanberry replied:
In procedural docs, we try to use present tense, because the action is actually happening at that moment.

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+
+:::info{title="Consequences of locking a domain"}
+
+Locking your domain affects all _enrolled_ users who login with an email address in the federated domain. The only way to unlock a domain is to remove it from Apple Business Manager, which until re-locking, will prevent them from accessing Apple services.
```
```suggestion
Locking your domain affects all _enrolled_ users who log in with an email address in the federated domain. The only way to unlock a domain is to remove it from Apple Business Manager, which until re-locking, will prevent them from accessing Apple services.
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+
+:::
+
+### Federated Authentication
```
```suggestion
### Federated authentication
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+
+Now that our authentication domain is verified, you're ready to configure federated authentication.
+
+Once more from from the Apple Business Manager dashboard, press your name on the sidebar, then select **Preferences**. From the preferences page, select **Managed Apple Accounts** tab, and press **Get Started** under the "User sign in and directory sync" section.
```
```suggestion
From from the Apple Business Manager dashboard, press your name on the sidebar, then select **Preferences**. From the preferences page, select **Managed Apple Accounts** tab, and press **Get Started** under the "User sign in and directory sync" section.
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+| SSF Config URL    | `[Your SSF Config URL]`    |
+| OpenID Config URL | `[Your OpenID Config URL]` |
+
+Press **Continue** to complete the setup. Apple will now verify the configuration and prompt you to test the authentication flow. During this test, Apple will redirect you to authentik to authenticate, and then back to Apple to complete the process. When the test is successful, press **Done** to complete the setup.
```
```suggestion
Press **Continue** to complete the setup. Apple will now verify the configuration and prompt you to test the authentication flow. During this test, Apple redirects you to authentik to authenticate, and then back to Apple to complete the process. When the test is successful, press **Done** to complete the setup.
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,295 @@
+
+:::tip{title="Troubleshooting"}
+
+If the test fails, double-check the values you've entered and make sure that the authentik instance is accessible from the internet. Check the authentik logs for any errors that might have occurred during the authentication process. Scope mappings are a common source of issues, so double-check that you've created the necessary mappings, and that they're all assigned to the provider.
```
Very nice and kind (and well-written) tip.

tanberry reviewed (changes requested):
Thank you so much for writing the ABM integration guide! I can imagine that a lot of setup and config and work.

I added quite a few edits, sorry, but most are simple things like following the Style Guide with sentence-style capitalization for titles, present tense, and (for now) using "click" instead of "press". There are some more complex edits; I'd suggest we set up a walk-through to go over any you want to discuss. @dominic-r  we can try to schedule it at a time you are available, though @GirlBossRush is in same time zone as me, so 6 (5?) hours ahead of you...

GirlBossRush commented:
@tanberry @dominic-r @BeryJu

Thank you again for reviewing this integration. It's definitely one of the more difficult integrations to run through, especially given the some of prerequisites on Apple's dashboard.

I've attempted to make this iteration closer to our style guide with some caveats:

- The disclaimer at the top about Business Manager vs School Manager is a unique aspect of the integration I haven't found too much clarity on. It may need revising.
- I've had some difficulty moving the "Create application" step into the "Apple Business Manager configuration" section since the reader needs to backtrack to retrieve their SSF Config URL. There might be a need for a new type of admonition (e.g. Get your text editor for this data cause you'll need it later)
- The recap section doesn't have any prior art. It feels like a reasonable pit-stop before embarking into Apple's labyrinthian dashboard. The checklist also saved me a few times while retesting the setup.
- The account capture disclaimers are admittedly wordy, much like Apple's own dashboard warnings. Prematurely performing the action can derail the integration and disrupt users in such a way that we really need to drive home how impactful this can be.
- The port number disclaimer is ergonomically awkward but a necessary step. This may be a future opportunity for us to make the Apple Business Manager setup flow a wizard.
- The testing flow may need a future revision that tests device enrollment, possibly with a hardware device we can fiddle with.
- The adjustments to the shared signals docs reflect my limited knowledge of their colloquial use and the potential arrangements they make possible. They may need revising.

tanberry commented on website/integrations/services/index.mdx:
```diff
@@ -10,7 +10,8 @@ import SupportBadge from "@site/src/components/SupportBadge";
 
-:::info{title="Support Levels"}
+:::info
+
```
@GirlBossRush I'd like to remove the edits for this file from this PR please, and let's chat about the changes. The big section about the the diff types of badges is too overwhelming, and I don't want to remove the section about Sources just yet, because they used to be here under Integrations, and so I want to link to them. I also want to look into the use of the cards for the Integration categories... I didn't even know that got changed.

GirlBossRush replied:
Reverted to include the title.

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,327 @@
+
+> Apple Business Manager is a web-based portal for IT administrators, managers, and procurement professionals to manage devices, and automate device enrollment.
+>
+> Organizations using Apple Business Essentials can allow their users to authenticate into their Apple devices >using their IdP credentials, typically their company email addresses.
```
```suggestion
Organizations using Apple Business Essentials can allow their users to authenticate into their Apple devices using their IdP credentials, typically their company email addresses.
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,327 @@
+
+Before proceeding with the Apple Business Manager configuration, you will to need to configure authentik to act as an identity provider for Apple. Start by logging into your authentik instance and navigating to the Admin interface.
+
+### 1. Scope mapping
```
When I mentioned numbered steps, I meant the actual steps below each main section. So starting with line 84., it wold be "1. In the Admin interface, navigate to...."

tanberry replied:
```suggestion
### Scope mapping
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,327 @@
+
+### 1. Scope mapping
+
+In the Admin interface, navigate to **Customization** -> **Property Mappings**.
```
```suggestion
1. In the Admin interface, navigate to **Customization** -> **Property Mappings**.
```

tanberry replied:
```suggestion
2. In the Admin interface, navigate to **Customization** -> **Property Mappings**.
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,327 @@
+
+In the Admin interface, navigate to **Customization** -> **Property Mappings**.
+
+Click **Create** and when prompted for a choice of mapping, select **Scope Mapping**.
```
```suggestion
3. Click **Create** and when prompted for a choice of mapping, select **Scope Mapping**.
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,327 @@
+
+Save the scope mapping and click **Create** to confirm the creation of all three scope mappings.
+
+### 2. Signing keys
```
```suggestion
### 2. Create signing keys
```

tanberry replied:
Best practice in tech writing: one wants a title to tell you what the section is about... we use "hintful" titles, such as "About signing keys" or "Create signing keys" to help the reader know what to expect in that secton.. is it conceptual (in which case many readers will skip over it haha), or is it procedural, and will tell them How To do something (which is probably what they are looking for).

dominic-r commented on website/docs/add-secure-apps/providers/ssf/index.md:
```diff
@@ -1,30 +1,45 @@
+The role of the transmitter is to send signals to the receiver, and the receiver is the application that listens for the signals. Each system receives and sends signals, but the signals are different.
+
+The identity provider is the transmitter of _identity signals_ through authentik to the application,
+and the application is the receiver of _identity signals_, such as user information, groups, and roles.
```
Can this newline get removed? It seems accidental

dominic-r commented on website/docs/add-secure-apps/providers/ssf/index.md:
```diff
@@ -1,30 +1,45 @@
+and the application is the receiver of _identity signals_, such as user information, groups, and roles.
+
+An SSF provider is the transmitter of _security signals_ through authentik to the application,
+and the application is the receiver of _security signals_, such as device information, user events, and security events.
```
Same as above

GirlBossRush commented on website/docs/users-sources/sources/social-logins/google/index.mdx:
```diff
@@ -7,10 +7,12 @@ tags: [integration, google, saml]
 
-:::info{title="authentik as a third-party IdP"}
+:::info authentik as a third-party IdP
 authentik can also be configuration to authenticate users into Google services.
 
-For more information, see the [Google Workspace Integration](../../../../../integrations/services/google/) guide.
+For more information, see the [Google Workspace Integration](../../../../../integrations/services/google/).
+
+:::
```
Style linter fixes.

GirlBossRush commented on website/docs/users-sources/sources/social-logins/google/workspace/index.md:
```diff
@@ -52,7 +52,7 @@ Keep a text-editor handy because we'll be copying and pasting values between the
 
 The following placeholders are used:
 
-- `authentik.company`: The Fully Qualified Domain Name of the authentik installation.
```
Linter + style guide fixes

GirlBossRush commented on website/src/css/custom.css:
```diff
@@ -207,7 +207,7 @@ body {
     & .messageText {
         stroke: var(--ifm-background-color) !important;
-        stroke-width: 10;
+        stroke-width: 4;
```
Fix for Mermaid rendering in Chrome.

GirlBossRush commented on website/src/css/custom.css:
```diff
@@ -220,3 +220,14 @@ body {
         display: none;
     }
 }
+
+.contains-task-list {
+    input[type="checkbox"] {
+        appearance: none;
+    }
+    input[type="checkbox"]::after {
+        content: "✓";
+        color: var(--ifm-color-success-dark);
+        display: inline-block;
+    }
+}
```
Fix for missing styles on Markdown checkboxes

dominic-r commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+
+:::info Apple Device Management Platforms
+
+Apple repackages their device management platform into three brands to cater to different audiences:
```
I'm not sure that " repackages" is the right word. Maybe "packages" or "divides"?

tanberry replied:
I'd go with "packages", but "repackages" works fine.

GirlBossRush commented on web/src/admin/providers/ssf/SSFProviderFormPage.ts:
```diff
@@ -62,7 +62,11 @@ export class SSFProviderFormPage extends BaseProviderForm<SSFProvider> {
                 <span slot="header"> ${msg("Protocol settings")} </span>
                 <div slot="body" class="pf-c-form">
-                    <ak-form-element-horizontal label=${msg("Signing Key")} name="signingKey">
+                    <ak-form-element-horizontal
+                        label=${msg("Signing Key")}
+                        name="signingKey"
+                        required
+                    >
```
Requested by @BeryJu

dominic-r commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+
+## Authentication Flow
+
+This sequence diagram shows a high-level flow between the user's MacBook, authentik, and Apple.
```
Does this have to be a MacBook? Are other devices supported? If so, we could generalize to "Apple device"

BeryJu replied:
In theory it should be any apple device but I have not personally tested this on other devices yet

dominic-r commented on website/integrations/services/apple/index.md:
````diff
@@ -0,0 +1,385 @@
+```
+
+In short, Apple Business Manager recognizes the email domain
+as a federated identity provider controlled by authentik. When a user signs in with their email address, Apple redirects them to authentik for authentication. Once authenticated, Apple enrolls the user's device and grants access to Apple services.
````
Is this newline intentional?

dominic-r commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+
+### Placeholders
+
+The following placeholders are used:
```
```suggestion
The following placeholders are used in this guide:
```
As per the template

dominic-r commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+
+The workflow to configure authentik as an identity provider for Apple Business Manager involves creating scope mappings, signing keys, a Shared Signals Framework provider, and an OIDC provider. Together, these components will handle the authentication flow and backchannel communication between authentik and Apple Business Manager.
+
+### 1. Create scope mappings
```
Usually, we don't number our headers

tanberry replied:
```suggestion
### Create scope mappings
```

dominic-r commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+
+#### 1.2. Provide read access for the scope mapping
+
+From the **Property Mappings** overview, click **Create**.
```
We should include the  full steps or path to access this page

tanberry replied:
true... we shouldn't assume they know how to get to the Property Mappings overview.

dominic-r commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+| Scope Name | `ssf.manage`                        |
+| Expression | `return {}`                         |
+
+Click **Finish** and confirm that all three new scope mappings are listed in the overview.
```
After each scope configuration, you tell the user to verify its existence. This line feels redundant. Perhaps only keeping this one and removing the two others?

dominic-r commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+
+You will need to create **Signing Key** to sign Security Event Tokens (SET). This key is used to both sign and verify the SETs that are sent between authentik and Apple Business Manager.
+
+From the Admin interface, navigate to **System** -> **Certificates** and clicking **Generate** or **Create**.
```
```suggestion
From the Admin interface, navigate to **System** -> **Certificates** and click **Generate** or **Create**.
```

dominic-r replied:
Should we only recommend generate? We con assure the key meets safe standards

dominic-r commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+
+Under the **Advanced protocol settings**, add the newly created scope mappings to the provider, as well as the default `profile` scope:
+
+- [x] `Apple Business Manager ssf.manage`
```
Why add check marks here? Wouldn't bullet pointe suffice

GirlBossRush commented:
@tanberry @BeryJu,

A user following this integration is in for quite a journey! The style guide recommends the following:

> - Try to order the documentation sections in the order that makes it easiest for the user to follow. That is, order the sections in the same order as the actual workflow used to accomplish the task.
> - Use headings (sub-titles) to break up long documents, and make it easier to find a specific section.
> 

I haven't found prior art in existing docs that makes use of the table of contents, but given the style guidelines, this iteration prefixes headings with numeric subsections. This gets us the ordered steps and allows Docusaurus to generate a table of contents 😄 


<img width="1470" alt="Screenshot 2025-03-10 at 20 52 33" src="https://github.com/user-attachments/assets/85e6844a-eaa5-4959-ac9d-5cd2c693f629" />

dominic-r commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+- [x] `Apple Business Manager ssf.read`
+- [x] `Apple Business Manager profile`
+- [x] `authentik default OAuth Mapping: OpenID 'profile'`
+
```
If the wizard says everything is ok,l think we should assume it's fire and not tell the user to check. As you wish

dominic-r commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+
+:::tip OpenID Configuration URL
+
+From the provider overview page, copy the URL in the **OpenID Configuration URL** field to your text editor. You'll need this when configuring Apple Business Manager.
```
Please include steps to reach this page

dominic-r commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+
+### 4. Create Shared Signals Framework provider
+
+While the OIDC provider handles the authentication flow, you'll need to create a [Shared Signals Framework provider](https://docs.goauthentik.io/docs/add-secure-apps/providers/ssf/) to handle the backchannel communication between authentik and Apple Business Manager.
```
Please use relative links

dominic-r commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+    - [x] Management access.
+2. Created a OIDC provider to handle the authentication flow.
+    - [x] Added the scope mappings to the provider, **as well as the default `profile` scope.**
+    - [x] Specified the `Redirect URIs/Origins` and `Signing Key`.
```
Ul fields should be bold as per the style guide

dominic-r commented on website/integrations/services/apple/index.md:
````diff
@@ -0,0 +1,385 @@
++https://authentik.company:443/application/o/abm/.well-known/openid-configuration
+```
+
+**This is a common source of configuration issues!**
````
I don't think this line is needed

tanberry replied:
Ah, I kinda like the "heads up" of it. Let;s leave it.

dominic-r commented on website/integrations/services/apple/index.md:
````diff
@@ -0,0 +1,385 @@
++https://authentik.company:443/.well-known/ssf-configuration/abm
+```
+
+#### 2. Add port to OpenID Config URL
````
Could ne merge both these steps in one?

dominic-r reviewed (dismissed):
Hi Teffen! Thank you for this very complete PR. Here are a few things I noticed

tanberry commented on website/docs/add-secure-apps/providers/ssf/create-ssf-provider.md:
```diff
@@ -41,7 +42,7 @@ The workflow to implement an SSF provider as a [backchannel provider](../../appl
 3. Define the settings for the application:
 
-    - **Name**: define a descriptive name ofr the application.
+    - **Name**: define a descriptive name of the application.
```
Thanks for catching this!

tanberry commented on website/docs/add-secure-apps/providers/ssf/index.md:
```diff
@@ -1,30 +1,45 @@
 ---
 
-Shared Signals Framework (SSF) is a common standard for sharing asynchronous real-time security signals and events across multiple applications and an identity provider. The framework is a collection of standards and communication processes, documented in a [specification](https://openid.net/specs/openid-sharedsignals-framework-1_0-ID3.html). SSF leverages the APIs of the application and the IdP, ...
+Shared Signals Framework (<abbr>SSF</abbr>) is a collection of standards for sharing asynchronous security signals and events across multiple applications and an identity provider. SSF leverages the APIs of the application and the IdP, using privacy-protected, secure webhooks.
```
@Teffen, as I indicated in an earlier review (apparently dismissed) I want to **leave the link to the spec where it is mentioned** in this doc. Please do not make any of the changes in this section. Actually, please remove this file from this PR. The PR was intended to create the ABM integration docs. If you'd like to change the SSF Provider docs (and the several other files now included in this PR), please open a separate PR. Let's practice good PR hygiene.

tanberry commented on website/docs/add-secure-apps/providers/ssf/index.md:
```diff
@@ -1,30 +1,45 @@
 ## About Shared Signals Framework
 
-In authentik, an SSF provider allows applications to subscribe to certain types of security signals (which are then translated into SETs, or Security Event Tokens) that are captured by authentik (the IdP), and then the application can respond to each event. In this scenario, authentik acts as the _transmitter_ and the application acts as the _receiver_ of the events.
+An SSF provider allows applications to subscribe to _security signals_, translating them into Security Event Tokens (<abbr>SET</abbr>) that are captured by the IdP. In this arrangement, authentik acts as an mediator from an external event _transmitter_, ultimately with a specific application acting as the _receiver_ of the events.
```
Please leave original wording. I wrote this to talk about SSF within authentik.

tanberry commented on website/docs/add-secure-apps/providers/ssf/index.md:
```diff
@@ -1,30 +1,45 @@
+Another example use case is when an application uses SSF to subscribe to authorization events because the application needs to know if a user changed their password in authentik. If a user changes their password in authentik, an event will be generated and sent to the application, which can then take action based on the content of the event.
 
-Let's look at a few details about using SSF in authentik.
+## SSF in authentik
```
We do not need this header. There is already a header, called "About using SSF in authentik". This is sufficient. And why do you want to remove line 25?

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+In short, Apple Business Manager recognizes the email domain
+as a federated identity provider controlled by authentik. When a user signs in with their email address, Apple redirects them to authentik for authentication. Once authenticated, Apple enrolls the user's device and grants access to Apple services.
+
+---
```
We don't use HRs anywhere else in the docs, so let's remove this one and then have a chat about where we would want to use them, consistently across all docs.

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+
+### 1. Create scope mappings
+
+#### 1.1. Mapping the user's profile information
```
I know I asked that the individual steps are numbered, but... we don't use this MS style of numbering in any other docs... for now let's number only the actual steps, and not the top-level sections (the super-steps).

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+
+#### 1.2. Provide read access for the scope mapping
+
+From the **Property Mappings** overview, click **Create**.
```
```suggestion
In the Admin interface, navigate to **Customization -> Property Mappings**, and then click **Create**.
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+
+#### 1.3. Provide management access for the scope mapping
+
+From the **Property Mappings** overview, click **Create**.
```
```suggestion
In the Admin interface, navigate to **Customization -> Property Mappings**, and then click **Create**.
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+
+### 2. Create signing keys
+
+You will need to create **Signing Key** to sign Security Event Tokens (SET). This key is used to both sign and verify the SETs that are sent between authentik and Apple Business Manager.
```
```suggestion
You will need to create a **Signing Key** to sign Security Event Tokens (SET). This key is used to both sign and verify the SETs that are sent between authentik and Apple Business Manager.
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+
+## authentik configuration
+
+The workflow to configure authentik as an identity provider for Apple Business Manager involves creating scope mappings, signing keys, a Shared Signals Framework provider, and an OIDC provider. Together, these components will handle the authentication flow and backchannel communication between authentik and Apple Business Manager.
```
```suggestion
The workflow to configure authentik as an identity provider for Apple Business Manager involves creating scope mappings, signing keys, a Shared Signals Framework provider, and an OIDC provider/ application pair. Together, these components will handle the authentication flow and backchannel communication between authentik and Apple Business Manager.
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+
+### 1. Create scope mappings
+
+#### 1.1. Mapping the user's profile information
```
```suggestion
The first step is to map the user's profile information.
```

tanberry replied:
Ah, I think after staring at this for a wile (and it is very complex steps and sub-steps), I think the best way to avoid MS style numbering is to use a simple sentence explainingg the substeps. Let's see how this looks.

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+#### 1.1. Mapping the user's profile information
+
+From the authentik Admin interface, navigate to **Customization** -> **Property Mappings** and click **Create**.
+When prompted for a choice of mapping, select **Scope Mapping** and continue with the following form values:
```
```suggestion

When prompted for a choice of mapping, select **Scope Mapping** and continue with the following form values:
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+
+#### 1.1. Mapping the user's profile information
+
+From the authentik Admin interface, navigate to **Customization** -> **Property Mappings** and click **Create**.
```
```suggestion
1. From the authentik Admin interface, navigate to **Customization** -> **Property Mappings** and click **Create**.
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+| Name       | `Apple Business Manager profile` |
+| Scope Name | `profile`                        |
+
+This scope mapping will also require an expression to map Apple Business Manager's name field into into separate given and family names. This expression assumes that the name is formatted with the given name first, followed by the family name, delimited by a space. Consider adjusting the expression to match the name format used in your organization.
```
```suggestion
3. Add an expression to map Apple Business Manager's name field into into separate given and family names. This expression assumes that the name is formatted with the given name first, followed by the family name, delimited by a space. Consider adjusting the expression to match the name format used in your organization.
```

tanberry commented on website/integrations/services/apple/index.md:
````diff
@@ -0,0 +1,385 @@
+}
+```
+
+Click **Finish** and confirm that new scope mapping is listed in the **Property Mappings** overview.
````
```suggestion
4. Click **Finish** and confirm that new scope mapping is listed in the **Property Mappings** overview.
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+
+Click **Finish** and confirm that new scope mapping is listed in the **Property Mappings** overview.
+
+#### 1.2. Provide read access for the scope mapping
```
```suggestion
Next, you need to provide read access for the scope mapping
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+
+From the **Property Mappings** overview, click **Create**.
+
+When prompted for a choice of mapping, select **Scope Mapping** and continue with the following form values:
```
```suggestion
2. When prompted for a choice of mapping, select **Scope Mapping** and continue with the following form values:
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+| Scope Name | `ssf.read`                        |
+| Expression | `return {}`                       |
+
+Click **Finish** and confirm that new scope mapping is listed in the **Property Mappings** overview.
```
```suggestion
3. Click **Finish** and confirm that new scope mapping is listed in the **Property Mappings** overview.
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+
+Click **Finish** and confirm that all three new scope mappings are listed in the overview.
+
+### 2. Create signing keys
```
```suggestion
### Create signing keys
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+
+Click **Finish** and confirm that new scope mapping is listed in the **Property Mappings** overview.
+
+#### 1.3. Provide management access for the scope mapping
```
```suggestion
#### Finally, you need to provide management access for the scope mapping.
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+#### 1.3. Provide management access for the scope mapping
+
+From the **Property Mappings** overview, click **Create**.
+When prompted for a choice of mapping, select **Scope Mapping** and continue with the following form values:
```
```suggestion
2. When prompted for a choice of mapping, select **Scope Mapping** and continue with the following form values:
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+
+:::
+
+From the authentik Admin interface, navigate to **Applications** -> **Providers** and click **Create**.
```
```suggestion
1. From the authentik Admin interface, navigate to **Applications** -> **Providers** and click **Create**.
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+:::
+
+From the authentik Admin interface, navigate to **Applications** -> **Providers** and click **Create**.
+In the creation wizard, select **OAuth2/OpenID Provider**. While you can use any name for the provider, we recommend using `Apple Business Manager` for clarity.
```
```suggestion
2. For the **Name** of the provider you can use any name; we recommend using `Apple Business Manager` for clarity.

3. For the **Provider Type** select **OAuth2/OpenID Provider**, and then click **Next**.

3. Provide the configuration settings. Be sure to select a Signing Key.

4. Click **Finish** to create and save the provider.
```

tanberry replied:
Do hey need to select a Signing Key here, @GirlBossRush ? Please remove if no...

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+From the authentik Admin interface, navigate to **Applications** -> **Providers** and click **Create**.
+In the creation wizard, select **OAuth2/OpenID Provider**. While you can use any name for the provider, we recommend using `Apple Business Manager` for clarity.
+
+#### 3.1. Use Apple's Redirect URL
```
```suggestion
#### Use Apple's Redirect URL
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+
+#### 3.1. Use Apple's Redirect URL
+
+Under **Protocol settings**, click **Add entry** in the **Redirect URIs/Origins** field and fill in the following value:
```
```suggestion
Under **Protocol settings**, click **Add entry** in the **Redirect URIs/Origins** field and fill in the following value:
```

tanberry replied:
Oh, wait, isn't this step still part of the "Create an OIDC Provider" section??  If so, let's not a create a new header... makes it look like an entirely separate step.

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+From the Admin interface, navigate to **System** -> **Certificates** and clicking **Generate** or **Create**.
+When choosing a **Common Name**, consider using a name that reflects the purpose of the key, such as `apple-business-manager`.
+
+### 3. Create OIDC provider
```
So everything in this section should be a series of steps, that reflects as closely as possible the exact steps that the user will take in the UI, as they create a provider. We don't need substeps... just the steps on each panel of the create modal.

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+
+:::
+
+#### 3.3. Assign scope mappings
```
```suggestion
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+
+#### 3.3. Assign scope mappings
+
+Under the **Advanced protocol settings**, add the newly created scope mappings to the provider, as well as the default `profile` scope:
```
```suggestion
**Advanced protocol settings**: add the newly created scope mappings to the provider, as well as the default `profile` scope:
```

tanberry commented on website/integrations/services/apple/index.md:
````diff
@@ -0,0 +1,385 @@
+https://gsa-ws.apple.com/grandslam/GsService2/acs
+```
+
+#### 3.2. Use the newly created signing key
````
```suggestion
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+
+#### 3.2. Use the newly created signing key
+
+Under **Protocol settings** in the **Signing Key** field, choose the certificate you've created earlier. This key will be used to sign the OpenID Connect tokens.
```
```suggestion
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+- [x] `Apple Business Manager profile`
+- [x] `authentik default OAuth Mapping: OpenID 'profile'`
+
+Click **Finish** and confirm that `Apple Business Manager` is listed in the provider overview.
```
```suggestion
4. Click **Finish** and confirm that `Apple Business Manager` is listed in the provider overview.
```

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+
+---
+
+## Configuration checklist
```
I do see the usefulness of  checklist but don't think it is needed here. We want the docs to be lean, and these steps repeat the steps above. And... there is still a LOT of content still to go. :-)

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+
+By default, authentik will generate the **SSF Config URL** and **OpenID Configuration URL** without the default HTTPS port (443). However, Apple requires the port number to be included when providing the URLs in the configuration.
+
+From your text editor, update the URLs to include the port number `443`:
```
?? Is there a command that they need to run in their...  by text editor do you mean their VSCode or whatever IDE they use?

tanberry commented on website/integrations/services/apple/index.md:
```diff
@@ -0,0 +1,385 @@
+
+### 1. Add and verify your domain
+
+From the [Apple Business Manager dashboard](https://business.apple.com/), click **your account name** on the sidebar, then select **Preferences**. From the preferences page, select **Managed Apple Accounts** tab, and click **Add Domain**.
```
Ideally, these are the steps that need to be numbered, so that readers staring at the ABM UI can walk through the steps easily, and recognize the exact fields in the UI. And, ideally, the steps (which follow the UI) are not broken up with the H4s like the 1.1 or 1.2 below.

Let's try to make this match the "authentik configuration" section (as far as numbering and following the UI) as much as possible, but... it doesn't have to be perfect. Let's just et this massive Guide merged. ;-)

tanberry reviewed (changes requested):
Thank you for this huge contribution @GirlBossRush! It's a major addition to our integration guides, and highlights our new SSF provider. Merci!

GirlBossRush committed:
website/docs: Flesh out Apple Business Manager integration.

- Fix missing required attribute.
- Adjust visible headers in table of contents.
- Fix Docusaurus warning for nested paragraphs.

GirlBossRush commented on website/docusaurus.config.ts:
```diff
@@ -71,7 +71,8 @@ const createConfig = (): Config => {
                 copyright: `Copyright © ${new Date().getFullYear()} Authentik Security Inc. Built with Docusaurus.`,
             },
             tableOfContents: {
-                maxHeadingLevel: 5,
+                minHeadingLevel: 2,
+                maxHeadingLevel: 3,
```
Requested by @tanberry 😄

tanberry reviewed (approved):
Wow and thank you, @GirlBossRush, this is a monumental integration guide. Well done!
